Week3/homework/step2-3.js

```diff
@@ -30,16 +29,24 @@ console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 function repeatStringNumTimesWithDoWhile(str, num) {
   // eslint-disable-next-line prefer-const
   let result = '';
-
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
-
+  do {
+    if (num === 3) {
+      result += str;
+    } else if (num === 2) {
+      result += str;
+    } else if (num === 1) {
+      result += str;
+    } else {
+      result = '';
+    }
+    num--;
+  } while (num > 0);
```

Comment:
Ah. I think this is not completely right. You are presuming that the value of `num` would always be 3 with this logic. But it is not in your control, is it? It can be of any value. Could you revisit this please?